File: OJ/usaco/cpp/humble.cpp
/*
ID:oifox20071
LANG:C++
PROG:humble
*/
#include <iostream>
#include <cstring>
using namespace std;
const int maxk = 100;
const int maxn = 100001;
const int infinity = 2000000000;
int prime[maxk], a[maxn], f[maxk];
int main()
{
	freopen("humble.in", "r", stdin); freopen("humble.out", "w", stdout);
	int n, k;
	cin >> k >> n;
	memset(f, 0, sizeof(f));
	for (int i = 0; i < k; ++ i)
		cin >> prime[i];
	a[0] = 1;
	for (int i = 1; i <= n; ++ i)
	{
		int min = infinity;
		for (int j = 0; j < k; ++ j)
			if (a[f[j]] * prime[j] < min)
				min = a[f[j]] * prime[j];
		a[i] = min;
		for (int j = 0; j < k; ++ j)
			if (a[f[j]] * prime[j] == min)
				++ f[j];
	}
	cout << a[n] << endl;
	return 0;
}

File: OJ/usaco/pascal/s3/s31/humble.pas
{
ID:lijianx1
PROG:humble
LANG:PASCAL
}
program humble;
const
	maxn = 100000;
	maxm = 100;
	infinity = 2000000000;
var
	a:array[0..maxn] of longint;
	prime,f:array[1..maxm] of longint;
	i,j,k,min,n:longint;
begin
	assign(input,'humble.in'); reset(input);
	readln(k,n);
	fillchar(f,sizeof(f),0);
	for i:=1 to k do read(prime[i]);
	close(input);
	a[0]:=1;
	for i:=1 to n do begin
		min:=infinity;
		for j:=1 to k do
			if (a[f[j]] * prime[j] < min) then min:=a[f[j]] * prime[j];
		a[i]:=min;
		for j:=1 to k do
			if (a[f[j]] * prime[j] = min) then inc(f[j]);
	end;
	assign(output,'humble.out'); rewrite(output);
	writeln(a[n]);
	close(output);
end.